common/data_refinery_common/job_lookup.py

```diff
@@ -23,13 +34,103 @@ class ProcessorPipeline(PipelineEnums):
 
 
 class DiscoveryPipeline(PipelineEnums):
+
     """Pipelines which discover appropriate processing for the data."""
     pass
 
 
 class Downloaders(Enum):
+
     """An enumeration of downloaders for downloader_task."""
     ARRAY_EXPRESS = "ARRAY_EXPRESS"
     SRA = "SRA"
     TRANSCRIPTOME_INDEX = "TRANSCRIPTOME_INDEX"
     GEO = "GEO"
+    NONE = "NONE"
+
+
+def _is_platform_supported(platform: str) -> bool:
+    upper_platform = platform.upper()
+
+    # Check if this is a supported Microarray platform.
+    for supported_platform in utils.get_supported_microarray_platforms():
+        if (supported_platform["platform_accession"].upper() == upper_platform
+                or supported_platform["external_accession"].upper() == upper_platform):
+            return True
+
+    # Check if this is a supported RNASeq platform.
+    # GEO RNASeq platform titles often have organisms appended to
+    # an otherwise recognizable platform. The list of supported
+    # RNASeq platforms isn't long, so see if any of them are
+    # contained within what GEO gave us.
+    # Example: GSE69572 has a platform title of:
+    # 'Illumina Genome Analyzer IIx (Glycine max)'
+    # Which should match 'Illumina Genome Analyzer IIx'
+    # because RNASeq platforms are organism agnostic.
+    for supported_platform in utils.get_supported_rnaseq_platforms():
+        # Spacing can be inconsistent, easiest to just remove it entirely.
+        if supported_platform.upper().replace(" ", "") in upper_platform.replace(" ", ""):
+            return True
+
+    return False
+
+
+def determine_downloader_task(sample_object: Sample) -> Downloaders:
+    if _is_platform_supported(sample_object.platform_accession_code):
```

Comment:
This seems like a very important function, so should a docstring describing its high-level purpose